Source/common/SNTFileInfo.m

```diff
@@ -572,6 +572,11 @@ - (NSDictionary *)embeddedPlist {
     NSData *cmdData = [self safeSubdataWithRange:NSMakeRange(offset, sz_segment)];
     if (!cmdData) return nil;
 
+    struct load_command *lc = (struct load_command *)[cmdData bytes];
+    if (lc->cmdsize < sizeof(struct load_command)) {
+      return nil;
+    }
```

Comment:
What's this for? `cmdData` was declared to be the size of either `segment_command_64` or `segment_command` and safeSubdataWithRange will return nil if there isn't enough data to read (which was checked on line 573)

Reply:
If the command isn't a segment load command, then we only increment `offset` by `lc->cmdsize` which the fuzzer would sometimes set to 0, causing us to re-parse the same command `ncmds` times (up to 2**32 times) eventually OOMing.

Reply:
Oh, I see now this was in the opening comment, sorry for not reading.